4.queueRange.js

```diff
@@ -3,42 +3,73 @@
 // and returns the range (difference between the minimum and maximum).
 // 
 // Example Queue class:
-// 
-// class Queue {
-//   constructor() {
-//     this.array = []
-//   }
-// 
-//   isEmpty() {
-//     return this.array.length === 0
-//   }
-// 
-//   peek() {
-//     if (this.isEmpty()) {
-//       return null
-//     } else {
-//       return this.array[0]
-//     }
-//   }
-//
-//   enqueue(value) {
-//     this.array.push(value)
-//   }
-//
-//   dequeue() {
-//     return this.array.shift()
-//   }
-// }
+
+class Queue {
+  constructor() {
+    this.array = []
+  }
+
+  isEmpty() {
+    return this.array.length === 0
+  }
+
+  peek() {
+    if (this.isEmpty()) {
+      return null
+    } else {
+      return this.array[0]
+    }
+  }
+
+  enqueue(value) {
+    this.array.push(value)
+  }
+
+  dequeue() {
+    return this.array.shift()
+  }
+}
 // 
 // Imagine the queue contains -145, 45, 11, 98
 // range(q) should return 243
 
 // Write your function here
-const range = (queue) => {
 
+
+// variable for range
+// grab largest num 
+// grab smallest num
+// subtract
+const range = (queue) => {
+    if (queue.isEmpty()) return;
+    let min = 0
+    let max = 0
```

Comment:
dont start min and max at 0. start at infinity and -infinity. If no negatives you'll have a false min otherwise, and vice versa for max.